zimports.py

```diff
@@ -14,6 +14,9 @@
 import sys
 import time
 
+from flake8.main.application import Application
```

Comment:
This is explicitly documented as not a stable API that will change

Reply:
can you point me to where it's documented as not stable?

Reply:
Looking at their read the doc https://flake8.pycqa.org/en/latest/internal/cli.html I see no mention of it

Reply:
Literally you're looking at docs for things that are only for internal usage.

Reply:
It's only documented for contributors and maintainers, not for outside usage

Reply:
Fair enough. it may be worth pointing that out at https://flake8.pycqa.org/en/latest/internal/index.html, since I interpreted that as the public interface, since everything else is not published in the docs.

Is anything usable in flake8 as a module?
I guess we could use a constraint in setup.py (anything from 3.7.0 to the current version seems to work for our purpose here), or as I mentioned, looking at the code that parses the per-file-ignores is fairly small and could be vendored. 
The rest of the flake8 usage is mainly for convenience, nothing that could not be done manually

Reply:
btw, thanks for looking at it. re-reading the comment I realized it may be read as hostile. it was not by intention, sorry if it seemed that way.

Reply:
> btw, thanks for looking at it. re-reading the comment I realized it may be read as hostile. it was not by intention, sorry if it seemed that way.

I don't think I read it that way. I'm sorry if my response came across likewise. I was in a rush to respond in between meetings.

> Is anything usable in flake8 as a module?

I had hopes to expose a public API at some point but have taken a back seat as maintainer in recent years (yikes). I think Anthony likes to release 1-2x a year so adding a public API isn't likely on his radar at the moment. I've also considered having some of this stuff that other tools might want to use pulled out into a library. As @zzzeek is familiar with, over-doing that can lead to an OpenStack problem where there are a million libraries that are very tiny that are a hassle to deal with. I also haven't received significant demand for that to happen so it hasn't, yet. I do think it's totally reasonable though

Reply:
I guess the best solution is to vendor the parsing and the check. Thanks for the feedback.